Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```diff
@@ -0,0 +1,325 @@
+#nullable enable
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+using Arrowgene.Ddon.Database;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartEquipGradeUpHandler : GameStructurePacketHandler<C2SCraftStartEquipGradeUpReq>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartEquipGradeUpHandler));
+        private static readonly List<StorageType> STORAGE_TYPES = new List<StorageType> {
+            StorageType.ItemBagConsumable, StorageType.ItemBagMaterial, StorageType.ItemBagEquipment, StorageType.ItemBagJob, 
+            StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest
+        };
+        private static readonly List<StorageType> StorageEquipNBox = new List<StorageType> {
+            StorageType.ItemBagEquipment, StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest, StorageType.CharacterEquipment
+        };
+
+        private readonly ItemManager _itemManager;
+        private readonly Random _random;
+
+        public CraftStartEquipGradeUpHandler(DdonGameServer server) : base(server)
+        {
+            _itemManager = Server.ItemManager;
+            _random = new Random();
+        }
+
+        public override void Handle(GameClient client, StructurePacket<C2SCraftStartEquipGradeUpReq> packet)
+        {
+            #region Initializing vars
```

Comment:
This function is too large. It seems we should break it up into smaller functions if the logic is this complex.

Reply:
This function is sitll way too large. The fact that you needed to use `#region` should be a hint of such. I think one way to reduce many lines here would to simply depending on the default values of the objects being valid. Instead of initializing everything to 0. They should already be 0 by default. For members which are other classes, they should be allocated in the constructor.